release/releaser.go

```diff
@@ -230,10 +231,43 @@ func (r *Releaser) releaseWithoutUpdate(method, msg string, inst *instance.Insta
 	return res, nil
 }
 
-func (r *Releaser) execute(inst *instance.Instance, actions []ReleaseAction, kind flux.ReleaseKind, updateJob func(string, ...interface{})) error {
+func (r *Releaser) execute(inst *instance.Instance, actions []ReleaseAction, kind flux.ReleaseKind, job *jobs.Job, updateJob func(string, ...interface{})) (err error) {
 	rc := NewReleaseContext(inst)
 	defer rc.Clean()
 
+	defer func() {
+		cfg, err2 := inst.GetConfig()
+		if err2 != nil {
+			if err == nil {
+				err = errors.Wrap(err2, "sending notifications")
+			}
+			return
+		}
+		// Filling this from the job is a temporary migration hack. Ideally all
+		// the release info should be stored on the release object in a releases
+		// table, and the job should really just have a pointer to that.
+		err2 = notifications.Release(cfg, flux.Release{
+			ID:        flux.ReleaseID(job.ID),
+			CreatedAt: job.Submitted,
+			StartedAt: job.Claimed,
+			// TODO: fetch the job and look this up so it matches
+			EndedAt:  time.Now().UTC(),
+			Done:     true,
+			Priority: job.Priority,
+			Status:   flux.ServiceReleaseStatus(job.Status),
```

Comment:
Should this be one of the enum values? Since we only use this release for the notification, and the notification doesn't use this field, it doesn't make a difference to the outcome. But it may avoid a slip-up when this code is adapted later.